src/main/java/org/HdrHistogram/AbstractHistogram.java

```diff
@@ -729,13 +729,10 @@ public void add(final AbstractHistogram otherHistogram) throws ArrayIndexOutOfBo
      */
     public void subtract(final AbstractHistogram otherHistogram)
             throws ArrayIndexOutOfBoundsException, IllegalArgumentException {
-        long highestRecordableValue = valueFromIndex(countsArrayLength - 1);
-        if (highestRecordableValue < otherHistogram.getMaxValue()) {
-            if (!isAutoResize()) {
-                throw new ArrayIndexOutOfBoundsException(
-                        "The other histogram includes values that do not fit in this histogram's range.");
-            }
-            resize(otherHistogram.getMaxValue());
+        if (highestEquivalentValue(otherHistogram.getMaxValue()) >
+                highestEquivalentValue(valueFromIndex(this.countsArrayLength - 1))) {
```

Comment:
Second try on this check; view with suspicion since I already messed it up once, but this makes sense to me: if the top of the bucket needed in _this_ histogram to represent the _other_ histogram's max is above the largest value _this_ histogram can represent, subtraction can't work.